mlflow/utils/requirements_utils.py

```diff
@@ -533,6 +552,14 @@ def _check_requirement_satisfied(requirement_str):
             requirement=requirement_str,
         )
 
+    if pkg_name == "mlflow" and "gateway" in req.extras:
+        try:
+            import mlflow.gateway
+        except ModuleNotFoundError:
+            return _MismatchedPackageInfo(
+                package_name=pkg_name, installed_version=None, requirement=requirement_str
```

Comment:
reusing the gateway detection logic from [here](https://github.com/mlflow/mlflow/blob/master/mlflow/__init__.py#L213-L216)

as mentioned in the PR description, i could maybe change `installed_version` here to be something like `mlflow=={version}` rather than `None`, would that be more helpful?